pkg/csi/service/vanilla/controller_helper.go

```diff
@@ -139,3 +141,23 @@ func convertCnsVolumeType(ctx context.Context, cnsVolumeType string) string {
 	}
 	return volumeType
 }
+
+func isVolIDInDeviceList(ctx context.Context, blockVolID string, nodeVM *vsphere.VirtualMachine) (bool, error) {
+	// Verify if the volume id is on the VM backing the virtual disk devices.
+	vDeviceListPerVM, err := nodeVM.Device(ctx)
```

Comment:
nodeVM.Device(ctx) calls VC for every node VM. You need get all the devices for all node VM's at once in a single VC call and use that to verify if the K8s volumes are present in node VM's device list.

Can you please change this logic.